app/models/asset.rb

```diff
@@ -86,7 +86,13 @@ class Asset
     after_transition to: :uploaded do |asset, _|
       asset.save!
       asset.remove_file!
-      FileUtils.rmdir(File.dirname(asset.file.path))
+      dir = File.dirname(asset.file.path)
+      begin
+        FileUtils.rmdir(dir)
+      rescue Errno::ENOTEMPTY
+        files = Dir["#{dir}/*"].map { |path| path.split("/").last }
+        raise "Attempted to delete non-empty directory #{dir}. Contents: #{files}"
```

Comment:
I think rather than raising a custom error, we should send the original exception to Notify but include the contents of the directory in the context.

See https://docs.sentry.io/clients/ruby/context/ for more detail (although `Raven.capture_event` is [replaced with `GovukError.notify`](https://github.com/alphagov/govuk_app_config/blob/56ae9cf05a1958fe41e0e8d53cb5ff918f82234b/lib/govuk_app_config/govuk_error.rb#L5))

Reply:
Nice idea! Have now implemented 👍